### PR TITLE
Replace unsafeAddr by baseAddr

### PR DESCRIPTION
**Pull request by lchenut**
It should fix some of the https://github.com/status-im/nim-protobuf-serialization CI problems.
The rest of the problems should be solved with https://github.com/status-im/nim-faststreams/issues/31

### CODE REVIEW

**Comment by arnetheduck** [comment]
hm, I'm not convinced `baseAddr` is well-defined for empty arrays - taking the address of an empty array is not meaningful _really_ - I guess using `baseAddr` points to these places more clearly, but we should probably revisit this at some point cc @zah (the risk being that with `baseAddr`, we turn a `Defect` into runtime-undetected UB).

The alternative solution here might be to avoid `copyMem` using something like `assign2`.

**Review by arnetheduck — APPROVED** [review]

**Comment by arnetheduck** [comment]
should also add a test for writing/finalizing with empty array

File: faststreams/buffers.nim [hunk 1]
```diff
@@ -335,7 +335,7 @@ template writeByte*(span: var PageSpan, val: byte) =
 
 template charsToBytes*(chars: openArray[char]): untyped =
   bind makeOpenArray
-  var charsStart = unsafeAddr chars[0]
```
Comment (arnetheduck) [inline faststreams/buffers.nim:338]: `baseAddr` has been fixed to return `ptr T` (so the `ptr byte` cast is no longer necessary) and to return nil when the source is empty, so this code will now be as safe as it gets

**Comment by arnetheduck** [comment]
devel failure unrelated cc @zah

**Comment by bung87** [comment]
where is `baseAddr` defined ？ I get  `.nimble\pkgs2\faststreams-0.3.0-62f7ac8fb200a8ecb9e6c63f5553a7dad66ae613\faststreams\outputs.nim(535, 34) Error: undeclared identifier: 'baseAddr'`

**Comment by arnetheduck** [comment]
it's part of `nim-stew` in `ptrops` - likely, you're using some old version

**Comment by bung87** [comment]
really? it seems all the status-im packages my project relys has no tag, so no version restrictions, I also checked nimble packages dir, all of them match the latest version declared in nimble file.

**Comment by bung87** [comment]
hmm, you're right, when not versioned nimble not trying getting the latest.
